lapce-data/src/document.rs

```diff
@@ -1364,20 +1365,31 @@ impl Document {
         self.apply_deltas(&deltas)
     }
 
-    pub fn styles(&self) -> Option<&Arc<Spans<Style>>> {
-        let styles = self
-            .semantic_styles
-            .as_ref()
-            .or_else(|| self.syntax().and_then(|s| s.styles.as_ref()));
-        styles
+    pub fn styles(&self) -> Option<Arc<Spans<Style>>> {
+        let semantic_styles = self.semantic_styles.as_ref();
+        let syntax_styles = self.syntax().and_then(|s| s.styles.as_ref());
+
+        if semantic_styles.is_some() && syntax_styles.is_some() {
+            let combined_styles = semantic_styles
+                .unwrap()
+                .merge(syntax_styles.unwrap(), |a, _| a.clone());
+
+            return Some(Arc::new(combined_styles));
+        } else if semantic_styles.is_some() {
+            return semantic_styles.map(|s| s.clone());
+        } else if syntax_styles.is_some() {
+            return syntax_styles.map(|s| s.clone());
+        }
+
+        None
```

Comment:
I feel like there is a better way to do this than cloning stuff every time. But I couldn't work out how to do it

Reply:
You're cloning an `Arc` so it's not that big a deal.